services/user/Workshop/ui/src/components/metadata-form.tsx

```diff
@@ -0,0 +1,271 @@
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Tagline</FormLabel>
+                <FormControl>
+                  <Input {...field} placeholder="Pets on a blockchain!" />
+                </FormControl>
+                <FormDescription>Brief description of the app.</FormDescription>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="icon"
+            render={() => (
+              <FormItem>
+                <FormLabel>Icon</FormLabel>
+                <FormControl>
+                  <div className="flex items-center space-x-4">
+                    <Input
+                      type="file"
+                      accept="image/*"
+                      onChange={handleIconChange}
+                      className="w-full max-w-xs"
+                    />
+                    {isIcon && (
+                      <img
+                        src={iconPreview || currentSrc}
+                        alt="Icon preview"
+                        className="h-10 w-10 rounded-lg object-cover"
+                      />
+                    )}
+                    {iconPreview && (
+                      <Button
+                        type="button"
+                        onClick={() => {
+                          setIconPreview(null);
+                          form.resetField("icon");
+                          form.resetField("iconMimeType");
+                        }}
+                        variant="outline"
+                      >
+                        <Trash className="w-4 h-4 text-muted-foreground" />
+                      </Button>
+                    )}
+                  </div>
+                </FormControl>
+
+                <FormDescription>
+                  Upload an icon for your app (recommended size: 512x512px)
+                </FormDescription>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+        </div>
+        <FormField
+          control={form.control}
+          name="longDescription"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Long Description</FormLabel>
+              <FormControl>
+                <Textarea
+                  placeholder="Raise, feed and trade your own digital pets!"
+                  className="resize-none"
+                  {...field}
+                />
+              </FormControl>
+
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <div className="md:grid grid-cols-3 gap-2">
+          <FormField
+            control={form.control}
+            name="tosSubpage"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Terms of service</FormLabel>
+                <FormControl>
+                  <Input placeholder="/terms-of-service" {...field} />
```

Comment:
I was a little confused by the placeholder in these three fields. My initial intuition was that they were defaults that would actually get submitted if I didn't overwrite them. Instead, `/` was submitted for them. 

I could see a couple ways of getting around that:

* Make them required so the user has to explicitly input values, or
* Make the placeholder value submit as the default value, unless overridden by the user

Reply:
Now explicit values are required.